src/hooks/useDebounce.tsx

```diff
@@ -0,0 +1,15 @@
+import { useState, useEffect } from "react";
+
+export const useDebounce = (value: string, delay: number): string => {
```

Comment:
You can use generic here, debounced value don't have to be string 

```
export const useDebounce = <T>(value: T, delay: number):  T => {...}
```

Watch https://www.youtube.com/watch?v=nViEqpgwxHE